### ISSUE TITLE
support CUSTOM_COMPILE_COMMAND

### ISSUE THREAD

**Issue by raylu** [astral-sh/uv#1527]
https://github.com/jazzband/pip-tools?tab=readme-ov-file#configuration supports `CUSTOM_COMPILE_COMMAND`, which replaces the command in the header. this is particularly useful if you use a private repo, because otherwise your header looks like
```
# This file was autogenerated by uv v0.1.2 via the following command:
#    uv pip compile --extra-index-url https://aws:[suuuuuuuuuuuuuuuuuuuper long password here]@[account].d.codeartifact.us-west-2.amazonaws.com/pypi/shared-packages-pypi/simple requirements.in --output-file requirements.txt
```

(adding a flag instead of the env var would be nice too)

**Comment by sergei-maertens** [comment]
+1 from me, we use a multi-environment approach to requirements files and the details are abstracted away in a helper shell script:

```bash
export CUSTOM_COMPILE_COMMAND="./bin/compile_dependencies.sh"
```

This information is now lost.

The output before was:

```
#
# This file is autogenerated by pip-compile with Python 3.9
# by the following command:
#
#    ./bin/compile_dependencies.sh
#
```

and has now turned into:

```
# This file was autogenerated by uv via the following command:
#    uv pip compile --output-file requirements/ci.txt requirements/base.txt requirements/test-tools.in
```

**Comment by astrojuanlu** [comment]
Closed in #2554?
